### DISCUSSION TITLE
Deprecation warning in Pydantic. The method "dict" in class "BaseModel" is deprecated.

### DISCUSSION THREAD

**Discussion by sachinwuds** [fastapi/fastapi#11912] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict
```

### Description

In fastapi documentation , the method "dict" in class "BaseModel" is deprecated
  The `dict` method is deprecated; use `model_dump` instead. 
Here is the link : https://fastapi.tiangolo.com/tutorial/body/#__tabbed_4_1
I also added screenshots please check.
<img width="913" alt="Screenshot 2024-07-29 at 6 53 11 PM" src="https://github.com/user-attachments/assets/623214c5-7cc8-4339-92d6-bd27c062b90c">
<img width="972" alt="Screenshot 2024-07-29 at 7 02 25 PM" src="https://github.com/user-attachments/assets/edf3918a-e500-45bf-b8f8-11799d552506">

### Operating System

macOS

### Operating System Details

Macbook Pro

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

3.11.7

**Comment by YuriiMotov** [comment, marked as answer]
I think that's because FastAPI supports both Pydantic v1 and Pydantic v2, but Pydantic v1 doesn't support `model_dump` method.

Although, I think it would be nice to have an inline comment like:

```
    item_dict = item.dict()  # Use "item.model_dump()" if you use Pydantic v2 
```

**Comment by tallcoleman** [comment]
There's another docs page that has an explanation box about the deprecation - perhaps this can be copied to this page as well?

https://fastapi.tiangolo.com/tutorial/extra-models/#multiple-models

<img width="696" height="173" alt="Screenshot 2025-08-24 at 2 53 56 PM" src="https://github.com/user-attachments/assets/42d2d735-4205-40f6-b9da-4426e3f225cc" />
